internal/xds/translator/translator.go

```diff
@@ -113,29 +114,57 @@ func Translate(ir *ir.Xds) (*types.ResourceVersionTable, error) {
 		tCtx.AddXdsResource(resource.ClusterType, xdsCluster)
 
 		// Search for an existing listener, if it does not exist, create one.
-		xdsListener := findXdsListener(tCtx, tcpListener.Address, tcpListener.Port)
+		xdsListener := findXdsListener(tCtx, tcpListener.Address, tcpListener.Port, core.SocketAddress_TCP)
 		if xdsListener == nil {
-			xdsListener = buildXdsListener(tcpListener.Name, tcpListener.Address, tcpListener.Port)
+			xdsListener = buildXdsTCPListener(tcpListener.Name, tcpListener.Address, tcpListener.Port)
 			tCtx.AddXdsResource(resource.ListenerType, xdsListener)
 		}
-
+		// todo: huabing zhao we should check the conflicted listeners: multiple TCP listeners or TCP and HTTP listeners
+		// should not listen on the same port
```

Comment:
Can you create an issue and xref in your comment?

Reply:
the xds code is complicated as is, prefer if we moved all these validations checks to higher levels such as IR or Gateway - API

Reply:
@danehans @arkodg So we agreed that all the validation should be in Gateway translator and assume the configuration passed down to XDS translator is correct?

Reply:
yeah I prefer that because that layer has the ability to signal back to the user using the `Status` and it performs a lot of the port conflict checks already today
https://github.com/envoyproxy/gateway/blob/9eecce661715577f9a95b0ec28d9a7450716d715/internal/gatewayapi/translator.go#L226